sw/tests/bare-metal/hostd/ethernet.c

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 ETH Zurich and University of Bologna.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Chaoqun Liang <chaoqun.liang@unibo.it>
+//
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include "printf.h"
+#include "util.h"
+   
+#define ETH_BASE       0x20000000
+
+#define MACLO_OFFSET                 0x0
+#define MACHI_OFFSET                 0x4
+#define IRQ_OFFSET                   0x10
+#define IDMA_SRC_ADDR_OFFSET         0x14
+#define IDMA_DST_ADDR_OFFSET         0x18
+#define IDMA_LENGTH_OFFSET           0x1c
+#define IDMA_SRC_PROTO_OFFSET        0x20
+#define IDMA_DST_PROTO_OFFSET        0x24
+#define IDMA_REQ_VALID_OFFSET        0x3c
+#define IDMA_REQ_READY_OFFSET        0x40
+#define IDMA_RSP_READY_OFFSET        0x44
+#define IDMA_RSP_VALID_OFFSET        0x48
+
+#define PLIC_BASE                    0x04000000
+#define RV_PLIC_PRIO19_REG_OFFSET    0x4c
+#define RV_PLIC_IE0_0_REG_OFFSET     0x2000
+#define RV_PLIC_IE0_0_E_19_BIT       19
+
+#define RV_PLIC_IP_0_OFFSET          0x1000
+
+#define PRINTF_ON
+
+
+int main(void) { 
+
+  if (hart_id() != 0) wfi();
+
+  #ifdef PRINTF_ON
+    printf ("Start test Ethernet...\n\r");
+  #endif 
+
+ *reg32(PLIC_BASE, RV_PLIC_PRIO19_REG_OFFSET) = 1;
+ *reg32(PLIC_BASE, RV_PLIC_IE0_0_REG_OFFSET)  |= (1 << (RV_PLIC_IE0_0_E_19_BIT)); // Enable interrupt number ;
```

Comment:
Use the [__base_plic](https://github.com/pulp-platform/cheshire/blob/main/sw/link/common.ldh#L46) for the PLIC base address.